### PR TITLE
Add runtime health endpoint

### LINKED ISSUE #21: Add runtime health endpoint
See https://github.com/apache/camel-k/issues/511

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #21

### CODE REVIEW

File: camel-k-runtime-health/src/main/java/org/apache/camel/k/health/HealthEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        @Override
+        protected void channelRead0(ChannelHandlerContext ctx, HttpRequest msg) throws Exception {
+            HttpResponseStatus status;
+            ByteBuf content;
+
+            if (!Objects.equals(path, msg.uri())) {
+                status = HttpResponseStatus.NOT_FOUND;
+                content = Unpooled.wrappedBuffer(KO);
+            } else if (context.getStatus() == ServiceStatus.Started) {
```
Comment (nicolaferraro) [inline camel-k-runtime-health/src/main/java/org/apache/camel/k/health/HealthEndpoint.java]: Is this true when routes are down for master-component/routepolicy?

Comment (lburgazzoli) [reply camel-k-runtime-health/src/main/java/org/apache/camel/k/health/HealthEndpoint.java]: yes, context is always run only the consumer/route is stopped

**Comment by lburgazzoli** [comment]
> Seems ok. I forsee issues when trying to add this to the camel k workflow. E.g. base images won't be used because an additional dependency is needed... Wdyt?

We may need to add an option to the publisher  to configure additional dependencies

**Comment by lburgazzoli** [comment]
@nicolaferraro what is not yet implemented is the integration with camel health's subsystem will do in a future pr

**Pull request merged**